CMakeLists.txt

```diff
@@ -47,6 +47,68 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+
+if(WIN32)
```

Comment:
I know you mentioned it as out-of-scope, but I agree that sharing this logic with macOS would be great, given that the vcpkg envs are structured practically identically as far as I understand.

Reply:
It would be great if you could create a follow-up PR for macOS.